### DISCUSSION TITLE
Using `state` and `send` as React props

### DISCUSSION THREAD

**Discussion by claidler** [statelyai/xstate#3425] [Q&A]
Is it expected and okay to use `state` and `send` as props in React to use in child components?

Like:

```typescript
   export const SomeComponent = () => {
       const [state, send] = useMachine(someMachine);
       return <SomeOtherComponent state={state} send={send} />
   }
```

And if so, is there a good way to define the types on these props? I have so far been importing Event and State types from xstate and doing it that way.

Thanks in advance

**Comment by Andarist** [comment]
> Is it expected and okay to use state and send as props in React to use in child components?

Is it okay? It depends on your architecture choices - it definitely increases the coupling between the parent and the child component but perhaps that's not a big deal in your situation. Just do whatever works for you best in your context.

> And if so, is there a good way to define the types on these props? I have so far been importing Event and State types from xstate and doing it that way.

You can use `StateFrom` and `InterpreterFrom`:
[TS playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgYygUwIYzQWQ8gC2ADs0AaOAZRizQDEoIQKBJY7KMdDhpuAXzgAzRiDgAiAB4BnGtnEBuOAHplcGAE8waaQC44AFgB0AZgBMRgIwAoa8gjFZcEPiKk4AXhTpaeQiTQACgRrOBQHbEkYfSR7AFd2NCh9AAYBMlC4aUI0FxjMsLQANzR2PURBDGkCsLgAH0Qa2trNbX1xOgB5TsUm2v4+hpDmkfUtNHaAIQBBACVe0bD+DKWM-gBKBVtWtDgABUYwaU9GsNlafWpaXhAAHh2IIWdXAIA+LbPSgBN9Ng4uNA8UT3caPZ7+UivADa4mk33EAF0tvwttYvmhkAAbDDocKOeBcCBHfQHInSVGEo5Gc7YIz2RJRLaqRa1AB6AH5bJTpNTvky1GzOdZubziF9gpkdu0uj01pthYceXCxRKwsyAAIwaQAWjQkm0yBguqgjCgkvG7QAqgA5ADS1s6AHVreI5VsgA)

**Reply by claidler** [reply]
Thanks @Andarist I didn't know about stateFrom and interpreterFrom. Really useful

**Comment by davidkpiano** [comment]
The more idiomatic approach would be to pass the _actor_ to your component:

```js
const SomeComponent = () => {
  const [state, send, actor] = useMachine(someMachine);

  return <SomeOtherComponent actor={actor} />
}
```

That way, you can have access to both `state` and `send` using `useActor(actor)`:

```js
const SomeOtherComponent = ({ actor }) => {
  const [state, send] = useActor(actor);
}
```

Also, if you're only passing the actor, you can use `useInterpret`:

```js
const SomeComponent = () => {
  const actor = useInterpret(someMachine);

  return <SomeOtherComponent actor={actor} />
}
```

**Reply by claidler** [reply]
I did not know you could do this! This is awesome. Thanks @davidkpiano
